### PR TITLE
EC-1738: Add unit tests for size label calculation

### PR DESCRIPTION
**Pull request by robnester-rh**
## Summary

Adds comprehensive unit tests and improves the action's compliance with GitHub Actions best practices.

### Changes

- **Separate entrypoint from testable logic** — `src/main.js` is the new ncc build entrypoint that calls `run()` unconditionally. `src/index.js` is a pure module exporting `getSizeLabel` and `run` with no side effects on import. This eliminates the `require.main === module` guard that ncc transforms unreliably.
- **Guard against non-PR events** — `run()` now fails with a clear message instead of a TypeError when triggered outside `pull_request`/`pull_request_target` events.
- **Tighten label filter** — `startsWith('size:')` narrowed to `startsWith('size: ')` to avoid matching labels that don't follow this action's format.
- **Add 23 tests** — 18 boundary tests for `getSizeLabel` across all six size tiers, plus 5 `run()` tests covering label add, label replace, no-op, non-PR event guard, and API error propagation.
- **Rebuild dist** — `dist/index.js` regenerated from new entrypoint.

### Review notes

- `dist/index.js` changes are mechanical rebuilds — sanity check only
- `package-lock.json` changes reflect upstream merges
- CodeRabbit local review: no findings (2 clean passes)
- Qodo self-review: all suggestions addressed or intentionally skipped

Ref: EC-1738

### CODE REVIEW

**Review by simonbaird — DISMISSED** [review]
Lgtm, but consider the Coderabbit suggestion.

**Review by dheerajodha — APPROVED** [review]
lgtm

File: src/index.js [hunk 1]
```diff
@@ -88,4 +97,8 @@ async function run() {
   }
 }
 
-run();
+module.exports = { getSizeLabel, run };
+
+if (require.main === module) {
+  run();
+}
```
[the remark this answers is not in the document]
Comment (robnester-rh) [reply src/index.js]: Yup, I addressed this in a subsequent commit.
